src/components/AnalysisPage/ThicknessAnalysisBox/ThickAnalysisBox.jsx

```diff
@@ -76,6 +77,7 @@ const ThickAnalysisBox = ({
         setFileNameFor3dModel("file.ply");
         const file = new File([blob], "file.ply", { type: blob.type });
         setFileFor3dModel(file);
+        setShowToggle(true);
       } else {
         if (listOfThicknessData.find((data) => data.id === data.id)) {
```

Comment:
if downloadPlyFilePlaceIn3dViewer fails, will setShowToggle(true) still run? or will this whole if statement just cancel?

I wasn't able to pass setShowToggle to downloadPlyFile to handle this in that function

Reply:
If downloadPlyFilePlaceIn3dViewer throws an error then setShowToggle(true) will not run. Throwing an error stops the code executing. When an error is thrown it 'bubbles up' until it goes up into a try in which case it will stop bubbling up and go into the catch block.

Heres an example of bubbling up:

```js
function top() {
  try {
    doThing()
  } catch(err) {
    console.log('catching the error, maybe there would be a UI popup or something')
  }
}

function lower() {
  lowest()
  console.log('2')
  otherThing()
}

function lowest() {
  console.log('1')
  throw new Error('a bad thing happened')
}


top()

```

This program logs '1' and then 'catching the error, maybe there would be a UI popup or something'.

It does not log '2'.

So  this shows an error bubbling up. As functions call other functions it creates new lower layers and so when one of those lower layers throws an error, it goes back up to the higher layers either until it goes into a try in which case it will stop bubbling up and go into the catch block or iif it never goes into a try block itll just bubble up until it gets to the top which would be probably the React component